### PR TITLE
Add paginator (first/last page)

### PR DESCRIPTION
**Pull request by anandsinghparihar**
#328 
Can we please add a First Page - Last Page in the history tab, once you click through to the end you have to click your way back to first page.

Could we also add the same at the top of the history section, this will save having to scroll down page every time in order to go to next or previous page.

### CODE REVIEW

**Comment by kewde** [comment]
Move the paginator to core-ui.
Get rid of the material imports here: https://github.com/anandsinghparihar/partgui/commit/2518f1a7505ed889de2486dc6a96611962786e55

**Comment by anandsinghparihar** [comment]
@gerlofvanek will create first and last page icon, pagination design is similar as it was only first page and last page button added. its call something like 

```
<app-paginator
        *ngIf="type == 'public' && getSinglePage().length > 0"
        [length]="getTotalCountForPagination()"
        [pageSize]="MAX_ADDRESSES_PER_PAGE"
        [pageSizeOptions]="PAGE_SIZE_OPTIONS"
        (page)="pageChanged($event)">
      </app-paginator>

```
history page list

**Comment by anandsinghparihar** [comment]
I guess the Only icon needs to be updated, let me check with CRZ if he completed first-page last page icon then I'll update it.

**Comment by anandsinghparihar** [comment]
Icon class change to
```
 chevron-single-left <
 chevron-single-right >
 chevron-double-left >>
 chevron-double-right <<
```

**Comment by anandsinghparihar** [comment]
@pciavald or @kewde  please review this PR. Icon is updated!!

**Comment by anandsinghparihar** [comment]
@allienworks Icon class is updated now this PR is ready for merge.

**Comment by allienworks** [comment]
Looks good to me – can someone pls review the code?

File: src/app/core-ui/paginator/paginator.component.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    this._emitPageEvent();
+  }
+
+  /**
+   * Whether there is a previous page.
+   * @return {?}
+   */
+  hasPreviousPage() {
+    return this.pageIndex >= 1 && this.pageSize !== 0;
+  };
```
Comment (pciavald) [inline src/app/core-ui/paginator/paginator.component.ts]: newline

File: src/app/core-ui/paginator/paginator.component.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+  getRangeLabel(page: number, pageSize: number, length: number) {
+    if (length === 0 || pageSize === 0) {
+      return '0 of ' + length;
+    }
+    length = Math.max(length, 0);
+    const startIndex = page * pageSize;
+    // If the start index exceeds the list length, do not try and fix the end index to the end.
+    const endIndex = startIndex < length ?
```
Comment (pciavald) [inline src/app/core-ui/paginator/paginator.component.ts]: ternary at beginning for readability

File: src/app/core-ui/paginator/paginator.component.ts (hunk 1 above)
Comment (pciavald) [reply src/app/core-ui/paginator/paginator.component.ts]: done

**Comment by pciavald** [comment]
I reviewed the code but I cannot review functionnality because of #532 : clean cloning your repo gives me the build error.

**Pull request merged**